fusion-client/src/main/java/au/com/dmg/fusion/client/FusionClient.java

```diff
@@ -340,6 +363,14 @@ private boolean validateMessage(SaleToPOI message) {
 					else if(tsResponse.getMessageReference() != null){ //Failure - In Progress
 						currentServiceID = tsResponse.getMessageReference().getServiceID();
 					}
+					else{ //Message not found, does not have MessageReference
```

Comment:
This verification is _not_ needed because this would have already been caught by the ServiceID verification _before_ the (messageCategory == MessageCategory.TransactionStatus) checking.

Reply:
That ServiceID verification checks for the MessageHeader ServiceID. This is for the case for a _"ErrorCondition" : "NotFound"_ TransactionStatusResponse. It will return the correct MessageHeader ServiceID, but will not have the MessageReference. If it doesn't check the else, it would proceed to comparing the lastMessageRefServiceID to the MessageHeader ServiceID.

Reply:
In this case, please place the lastMessageRefServiceID and messageValid checking inside an _if statement_ instead, which verifies if currentServiceID has been set (not null and not empty) - don't need to verify the validity of the ServiceID in the header again.

if((currentServiceID != null) && ( currentServiceID != ""))
{
_messageValid = lastMessageRefServiceID.equals(currentServiceID);
if(!messageValid){
      LOGGER.info("Unexpected Message Reference ServiceID " + currentServiceID + " received in " + messageCategory + ".  Expected value is " + lastTxnServiceID + " .  Will process the next message instead.");_
}